frontend/src/scenes/ingestion/ingestionLogic.ts

```diff
@@ -222,14 +282,58 @@ export const ingestionLogic = kea<ingestionLogicType>([
         setFramework: ({ framework }) => {
             eventUsageLogic.actions.reportIngestionSelectFrameworkType(framework)
         },
+        sidebarStepClick: ({ index }) => {
+            switch (index) {
+                case 0:
+                    actions.setPlatform(null)
+                    return
+                case 1:
+                    if (values.platform) {
+                        actions.setVerify(false)
+                        actions.setPlatform(values.platform)
+                    }
+                    return
+                case 2:
+                    if (values.platform) {
+                        actions.setVerify(true)
+                    }
+                    return
+                case 3:
+                    if (values.platform) {
+                        actions.setAddBilling(true)
+                    }
+                    return
+                default:
+                    return
+            }
+        },
```

Comment:
I appreciate you're not trying to refactor the whole thing but this definitely feels bug-prone. I wonder if it wouldn't be better as a simple enum. As only one screen can be shown anyways it definitely feels like one value rather than many.

But as I say, this isn't a refactor so fine to keep with the existing derivation approach.